File: app/src/main/java/com/anthonymandra/rawdroid/data/MetadataResult.java
package com.anthonymandra.rawdroid.data;

import android.arch.persistence.room.Relation;
import android.arch.persistence.room.TypeConverter;
import android.arch.persistence.room.TypeConverters;

import java.util.Arrays;
import java.util.List;

@TypeConverters({MetadataResult.class})
public class MetadataResult extends MetadataEntity {
//    public String keyword;
    public List<String> keywords;
    /**
     * This is a join from {@link FolderEntity}
     */
    // Relation requires set which is unnecessary
//    @Relation(
//        parentColumn = "parentId",
//        entityColumn = "documentUri",
//        entity = FolderEntity.class)
    public String parentUri;

    @TypeConverter
    public List<String> fromGroupConcat(String keywords) {
        return Arrays.asList(keywords.split(","));
    }
}

File: app/src/main/java/com/anthonymandra/rawdroid/data/MetadataResult.kt
package com.anthonymandra.rawdroid.data

import androidx.room.TypeConverter
import androidx.room.TypeConverters

@TypeConverters(MetadataResult::class)
class MetadataResult : MetadataEntity() {
    var keywords: List<String>? = null

    @TypeConverter
    fun fromGroupConcat(keywords: String?): List<String>? = keywords?.split(",")

    override fun equals(other: Any?): Boolean {
        if (this === other) return true
        if (javaClass != other?.javaClass) return false
        if (!super.equals(other)) return false

        other as MetadataResult

        if (keywords != other.keywords) return false

        return true
    }
}
